ddpui/api/transform_api.py

```
@@ -751,3 +765,590 @@ def post_unlock_canvas(request, payload: LockCanvasRequestSchema):
+                output_cols=output_cols,
+                dbtmodel=dbtmodel,
+            )
+
+            # create the edge from the terminal operation node to the model node
+            CanvasEdge.objects.create(from_node=terminal_node, to_node=model_node, seq=1)
+
+        logger.info(f"V2 operation node terminated successfully: {node_uuid}")
+        return convert_canvas_node_to_frontend_format(model_node)
+
+    except CanvasNode.DoesNotExist:
+        logger.error(f"Operation node {node_uuid} not found")
```

Ruff (0.14.5)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)